controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/JobStatus.java

```diff
@@ -94,6 +92,11 @@ public boolean isSuccess() {
     /** The error of the last completion, or empty if the last run succeeded */
     public Optional<DeploymentJobs.JobError> jobError() { return jobError; }
 
+    /** Returns whether this last failed on out of capacity */
+    public boolean isOutOfCapacity() {
+        return jobError.filter(error -> error == DeploymentJobs.JobError.outOfCapacity).isPresent();
```

Comment:
Argh, `Optional`s. I never managed to decide which is better: the above, or `return jobError.equals(Optional.of(DeploymentJobs.JobError.outOfCapacity));`.

Reply:
I'll probably replace the job error with a job status some time soon, and then this won't be an issue :)

Reply:
Yes, it's not pretty, that's why I added a method for it. I think the above is better as your suggestion creates an unnecessary object.